### PR TITLE
[MIG][11.0] hr_holidays: migration scripts

### PR DESCRIPTION
**Pull request by cubells**
cc @Tecnativa

### CODE REVIEW

**Comment by novawish** [comment]
I'm just curious: Is this issues (https://github.com/OCA/OpenUpgrade/issues/1380) fixed in here?

**Comment by pedrobaeza** [comment]
That's the idea, but right now, it seems that there are no changes, so it can be 2 things:

* We have to do something, as the red Travis suggests.
* There's something bad on the DB of the issue.

**Comment by cubells** [comment]
@novawish Yes, it is.

Now, this pr fix https://github.com/OCA/OpenUpgrade/issues/1380

File: addons/hr_holidays/migrations/11.0.1.5/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---Fields in module 'hr_holidays'---
+hr_holidays  / hr.employee              / leaves_count (integer)        : type is now 'float' ('integer')
+# Nothing to do
+
+hr_holidays  / hr.holidays              / first_approver_id (many2one)  : NEW relation: hr.employee
+# Done: renamed field
```
Comment (pedrobaeza) [inline addons/hr_holidays/migrations/11.0.1.5/openupgrade_analysis_work.txt]: You shouldn't rename the manager field, but:

* Add the field in pre.
* Copy the values from manager_id in pre.

File: addons/hr_holidays/migrations/11.0.1.5/openupgrade_analysis_work.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+hr_holidays  / hr.holidays              / manager_id2 (many2one)        : was renamed to second_approver_id [nothing to do]
```
Comment (pedrobaeza) [inline addons/hr_holidays/migrations/11.0.1.5/openupgrade_analysis_work.txt]: Always remember that until v12, you have to rename the field also although it has `oldname` attribute

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by cubells** [comment]
done @pedrobaeza

File: addons/hr_holidays/migrations/11.0.1.5/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+field_renames = [
+    ('hr.holidays', 'hr_holidays', 'manager_id2', 'second_approver_id')
+]
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    openupgrade.rename_fields(env, field_renames)
+    openupgrade.logged_query(
```
Comment (pedrobaeza) [inline addons/hr_holidays/migrations/11.0.1.5/pre-migration.py]: You should use for better completion the new openupgradelib method `add_fields`: https://github.com/OCA/openupgradelib/blob/7d9650376e1b64838b6caf2e90581176ac683808/openupgradelib/openupgrade.py#L1815

I'm adding it already

Comment (cubells) [reply addons/hr_holidays/migrations/11.0.1.5/pre-migration.py]: Perfect @pedrobaeza 

I didn't know this method.

Comment (pedrobaeza) [reply addons/hr_holidays/migrations/11.0.1.5/pre-migration.py]: It was added recently by me as I observed that there was a gap (the created XML-ID) and to try to simplify which PG type to use for adding the column.

On v12 won't be needed for the XML-ID part.

**Review by StefanRijnhart — APPROVED** [review]
What a sneaky bug in Odoo, to rename a field that is still defined...

**Pull request merged**